src/rules.jl

```diff
@@ -206,13 +206,26 @@ DNERule(args...) = DNE()
 #####
 
 """
-TODO
+    WirtingerRule([𝒟::Type, ]P::AbstractRule, C::AbstractRule)
+
+Construct a `WirtingerRule` object, which is an `AbstractRule` that consists of
+an `AbstractRule` for both the primal derivative ``∂/∂x`` and the conjugate
+derivative ``∂/∂x̅``. If the domain `𝒟` is specified, return a `Rule` evaluating
+to `P(Δ) + C(Δ)` if `𝒟 <: Real`, otherwise return `WirtingerRule(P, C)`.
 """
 struct WirtingerRule{P<:AbstractRule,C<:AbstractRule} <: AbstractRule
     primal::P
     conjugate::C
 end
 
+function WirtingerRule(𝒟::Type, P::AbstractRule, C::AbstractRule)
+    if 𝒟 <: Real || eltype(𝒟) <: Real
```

Comment:
We should make sure that the compiler elides this check when these are "statically" computable (it should)

Reply:
You mean by adding `@pure` or making it a generated function? I thought about it, but this seems fairly trivial and `@code_warntype` says it's fine. Still, probably wouldn't hurt.

Reply:
> You mean by adding @pure or making it a generated function?

Nope, the compiler should do this without any tricks, but type inference can be fairly conservative w.r.t. compositions of predicate statements it will choose to compute statically. This one definitely should be fine, but was worth checking 🙂 

> @code_warntype says it's fine

Awesome! Can we add some `@inferred` tests?

Reply:
> Awesome! Can we add some `@inferred` tests?

Will do